src/codegen/doc/mod.rs

```diff
@@ -517,13 +561,34 @@ where
             }
         }
 
-        if let Some(ref doc) = fn_.ret().doc {
+        let out_parameters: Vec<_> = no_array_length_params
+            .iter()
+            .filter(|param| {
+                param.direction == ParameterDirection::Out
+                    && !(param.name == "error" && param.c_type == "GError**")
+            })
+            .collect();
+
+        if fn_.ret().doc.is_some() || !out_parameters.is_empty() {
             writeln!(w, "\n# Returns\n")?;
-            writeln!(
-                w,
-                "{}",
-                reformat_doc(&fix_param_names(doc, &self_name), &symbols, &parent_name)
-            )?;
+            if let Some(ref doc) = fn_.ret().doc {
+                writeln!(
+                    w,
+                    "{}",
+                    reformat_doc(&fix_param_names(doc, &self_name), &symbols, &parent_name)
+                )?;
+            }
+            // fold out parameters as part of returns
+            for parameter in out_parameters {
+                if let Some(ref doc) = parameter.doc {
+                    writeln!(w, "## `{}`", nameutil::mangle_keywords(&parameter.name[..]))?;
```

Comment:
Still a bit wary of this name, it doesn't show up in the tuple making it slightly confusing where this is coming from? Adding the index like `` ## {i}: `{name}` `` probably isn't exactly pretty either.

Reply:
I don't think the indices adds any values nor it makes it clearer. The situation can somehow be imported but would require a bunch of special casing (whether it's an Option, Result.....) which is something not in the scope of this PR.

Reply:
It's not, we should probably leave this and let it to the user to figure out how to interpret the docs. Indeed, not the scope of this PR and probably not something that's really bothersome or fixable.